jax/_src/lax/lax.py

```diff
@@ -4519,25 +4519,43 @@ def _dilate_shape(shape, dilation):
 def _ceil_divide(x1, x2):
   return -np.floor_divide(np.negative(x1), x2)
 
+
+jax_enum_names = [*[es.name for es in xla_client.PaddingType], 'PYTORCH_SAME']
+JaxPaddingType = enum.Enum(
+    'JaxPaddingType',
+    jax_enum_names,
+    start=0,
+)
+
 def padtype_to_pads(in_shape, window_shape, window_strides, padding):
   """Convert padding string to list of pairs of pad values."""
-  PaddingType = xla_client.PaddingType
-
   if isinstance(padding, str):
-    mapping = {'VALID': PaddingType.VALID, 'SAME': PaddingType.SAME}
+    mapping = {
+        'VALID': JaxPaddingType.VALID,
+        'SAME': JaxPaddingType.SAME,
+        'PYTORCH_SAME': JaxPaddingType.PYTORCH_SAME,
+    }
     try:
       padding = mapping[padding.upper()]
     except KeyError as err:
       msg = "Unrecognized padding type: expected 'VALID' or 'SAME', got {}."
       raise RuntimeError(msg.format(padding)) from err
 
-  if padding == PaddingType.SAME:
+  if padding == JaxPaddingType.SAME:
     out_shape = _ceil_divide(in_shape, window_strides)
-    pad_sizes = np.maximum(0, (out_shape - 1) * window_strides +
-                                window_shape - in_shape)
+    pad_sizes = np.maximum(0, (out_shape - 1) * window_strides + window_shape -
+                           in_shape)
     return [(pad_size // 2, pad_size - pad_size // 2) for pad_size in pad_sizes]
-  elif padding == PaddingType.VALID:
+  elif padding == JaxPaddingType.VALID:
     return [(0, 0)] * len(in_shape)
+  elif padding == JaxPaddingType.PYTORCH_SAME:
+    out_shape = _ceil_divide(in_shape, window_strides)
+    pad_sizes = np.maximum(
+        0,
+        (out_shape - 1) * window_strides + window_shape - in_shape,
+        )
+    return [((pad_size + 1) // 2, pad_size - (pad_size + 1) // 2)
+            for pad_size in pad_sizes]
```

Comment:
This seems very very similar to the `SAME` case, so perhaps consider merging them? Wouldn't this code work?
```
if padding == VALID:
  return [(0, 0)] * len(in_shape)
else:
  out_shape = ...
  pad_sizes = ...
  half_pad_sizes = (pad_size + (padding == PYTORCH_SAME)) // 2
  return [(hps, pad_size - hps) for hps in half_pad_sizes]
```